### PR TITLE
[service.fussballpopup] 0.3.9

### PR DESCRIPTION
**Pull request by tyl0re**
### Description
This Servis Alows to Define Games via Programm Addon, .You Can Setup Multible Games, Or entiere COntensts. When a Games is Running the Results like Goal, ore Player Changig are Shown as an Overlay Message

### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [sciprt.foo.bar] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: service.fussballpopup/menu.py [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+
+profile    = xbmc.translatePath( __addon__.getAddonInfo('profile') ).decode("utf-8")
+temp       = xbmc.translatePath( os.path.join( profile, 'temp', '') ).decode("utf-8")
+translation = __addon__.getLocalizedString
+
+popupaddon=xbmcaddon.Addon("service.popwindow")
+popupprofile    = xbmc.translatePath( popupaddon.getAddonInfo('profile') ).decode("utf-8")
+popuptemp       = xbmc.translatePath( os.path.join( popupprofile, 'temp', '') ).decode("utf-8")
+  
+icon = xbmc.translatePath(xbmcaddon.Addon().getAddonInfo('path')+'/icon.png').decode('utf-8')
```
Comment (enen92) [inline service.fussballpopup/menu.py]: please use os.path.join()
`xbmc.translatePath(os.path.join(xbmcaddon.Addon().getAddonInfo('path'),'icon.png'))`

File: service.fussballpopup/menu.py [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+
+
+        
+def debug(content):
+    log(content, xbmc.LOGDEBUG)
+    
+def notice(content):
+    log(content, xbmc.LOGNOTICE)
+
+def log(msg, level=xbmc.LOGNOTICE):
```
Comment (enen92) [inline service.fussballpopup/menu.py]: default log level should be set to DEBUG. Only use the others if needed.

File: service.fussballpopup/menu.py [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+    text = text.replace('\u00f6', "ö")      
+    text = text.replace('\/', "/")    
+    #text = text.replace('\n', "")    
+    text = text.strip()
+    return text
+
+def addDir(name, url, mode, iconimage, desc="",sortname=""):
+    u = sys.argv[0]+"?url="+urllib.quote_plus(url)+"&mode="+str(mode)+"&name="+urllib.quote_plus(name)
+    ok = True
+    liz = xbmcgui.ListItem(name, iconImage=icon, thumbnailImage=iconimage)
```
Comment (enen92) [inline service.fussballpopup/menu.py]: as usual, iconImage and thumbnailImage are deprecated, use setArt instead. Also, please make an effort to keep this in mind for future submissions :)

File: service.fussballpopup/menu.py [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+    liz.setInfo(type="Video", infoLabels={"Title": name, "Plot": desc,"TrackNumber":sortname})  
+    if not iconimage or iconimage==icon or iconimage==defaultThumb:
+        iconimage = defaultBackground
+        liz.setProperty("fanart_image", iconimage)
+    else:
+        liz.setProperty("fanart_image", defaultBackground)
```
Comment (enen92) [inline service.fussballpopup/menu.py]: same for fanart image

File: service.fussballpopup/menu.py [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+    ok = xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]), url=u, listitem=liz, isFolder=True)
+    return ok
+  
+def addLink(name, url, mode, iconimage, duration="", desc="", genre=''):
+    u = sys.argv[0]+"?url="+urllib.quote_plus(url)+"&mode="+str(mode)
+    ok = True
+    liz = xbmcgui.ListItem(name, iconImage=defaultThumb, thumbnailImage=iconimage)
```
Comment (enen92) [inline service.fussballpopup/menu.py]: same here

File: service.fussballpopup/menu.py [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+    liz.setInfo(type="Video", infoLabels={"Title": name, "Plot": desc, "Genre": genre})    
+    liz.setProperty('IsPlayable', 'true')
+    liz.addStreamInfo('video', { 'duration' : duration })
+    liz.setProperty("fanart_image", iconimage)
```
Comment (enen92) [inline service.fussballpopup/menu.py]: and here

File: service.fussballpopup/menu.py [hunk 7]
```diff
@@ -0,0 +1,310 @@
...
+   day=struktur["current_matchday"]   
+   debug("Liega Day :"+ day)
+   debug("url day "+"https://api.sport1.de/api/sports/matches-by-season/co"+lieganr+"/se/")
+   content=geturl("https://api.sport1.de/api/sports/matches-by-season/co"+lieganr+"/se/")
+   struktur = json.loads(content)    
+   tage=struktur["round"]
+   
+   filename       = xbmc.translatePath( os.path.join( temp, 'spiel.txt') ).decode("utf-8")
+   if xbmcvfs.exists(filename) :
+      fp=open(filename,"r")
```
Comment (enen92) [inline service.fussballpopup/menu.py]: please use with statement

File: service.fussballpopup/menu.py [hunk 8]
```diff
@@ -0,0 +1,310 @@
...
+    debug("savespiel start")
+    filename       = xbmc.translatePath( os.path.join( temp, 'spiel.txt') ).decode("utf-8")
+    if xbmcvfs.exists(filename) :
+      fp=open(filename,"r") 
+      content=fp.read()
+      fp.close()   
+      content=content+"\n"+zeile      
+    else :
+       content=zeile   
+    fp = open(filename, 'w')
```
Comment (enen92) [inline service.fussballpopup/menu.py]: use with statement

File: service.fussballpopup/menu.py [hunk 9]
```diff
@@ -0,0 +1,310 @@
...
+def savemessage(message,image,grey,lesezeit)  :
+    message=unicode(message).encode('utf-8')
+    image=unicode(image).encode('utf-8')
+    debug("message :"+message)
+    debug("image :"+image)
+    debug("grey :"+grey)
+    debug("popuptemp :"+popuptemp)
+    debug("lesezeit :"+str(lesezeit))
+    filename=md5.new(message).hexdigest()  
+    f = open(popuptemp+"/"+filename, 'w')
```
Comment (enen92) [inline service.fussballpopup/menu.py]: use with

File: service.fussballpopup/resources/language/English/strings.po [hunk 10]
```diff
@@ -0,0 +1,265 @@
...
+msgctxt "#30055"
+msgid "International Copetition"
+msgstr ""
+
+msgctxt "#30056"
+msgid "Internationale League and Cup competitions"
+msgstr ""
+
+msgctxt "#30057"
+msgid "Nationale League"
```
Comment (enen92) [inline service.fussballpopup/resources/language/English/strings.po]: National

File: service.fussballpopup/resources/language/English/strings.po [hunk 11]
```diff
@@ -0,0 +1,265 @@
...
+msgctxt "#30054"
+msgid "International Club Contests"
+msgstr ""
+
+msgctxt "#30055"
+msgid "International Copetition"
+msgstr ""
+
+msgctxt "#30056"
+msgid "Internationale League and Cup competitions"
```
Comment (enen92) [inline service.fussballpopup/resources/language/English/strings.po]: International

File: service.fussballpopup/resources/language/English/strings.po [hunk 12]
```diff
@@ -0,0 +1,265 @@
...
+msgctxt "#30053"
+msgid " endet with"
+msgstr ""
+
+msgctxt "#30054"
+msgid "International Club Contests"
+msgstr ""
+
+msgctxt "#30055"
+msgid "International Copetition"
```
Comment (enen92) [inline service.fussballpopup/resources/language/English/strings.po]: Competition

File: service.fussballpopup/resources/language/English/strings.po [hunk 13]
```diff
@@ -0,0 +1,265 @@
...
+msgctxt "#30051"
+msgid " has started. The Result is "
+msgstr ""
+
+msgctxt "#30052"
+msgid " has started"
+msgstr ""
+
+msgctxt "#30053"
+msgid " endet with"
```
Comment (enen92) [inline service.fussballpopup/resources/language/English/strings.po]: ended

File: service.fussballpopup/resources/language/English/strings.po [hunk 14]
```diff
@@ -0,0 +1,265 @@
...
+msgctxt "#30048"
+msgid "The first Part of the Game"
+msgstr ""
+
+msgctxt "#30049"
+msgid " The Game "
+msgstr ""
+
+msgctxt "#30050"
+msgid " has endet. The Result is "
```
Comment (enen92) [inline service.fussballpopup/resources/language/English/strings.po]: has ended

File: service.fussballpopup/resources/language/English/strings.po [hunk 15]
```diff
@@ -0,0 +1,265 @@
...
+msgstr ""
+
+#empty string with id 30046
+
+msgctxt "#30047"
+msgid "The Second Part of the Game"
+msgstr ""
+
+msgctxt "#30048"
+msgid "The first Part of the Game"
```
Comment (enen92) [inline service.fussballpopup/resources/language/English/strings.po]: first half

File: service.fussballpopup/resources/language/English/strings.po [hunk 16]
```diff
@@ -0,0 +1,265 @@
...
+#empty string with id 30044
+
+msgctxt "#30045"
+msgid "The first Half of the  extra time"
+msgstr ""
+
+#empty string with id 30046
+
+msgctxt "#30047"
+msgid "The Second Part of the Game"
```
Comment (enen92) [inline service.fussballpopup/resources/language/English/strings.po]: second half

File: service.fussballpopup/resources/language/English/strings.po [hunk 17]
```diff
@@ -0,0 +1,265 @@
...
+msgctxt "#30022"
+msgid "Height of the Image Left"
+msgstr ""
+
+msgctxt "#30023"
+msgid "Width of the Image Right"
+msgstr ""
+
+msgctxt "#30024"
+msgid "Hight of the Image Right"
```
Comment (enen92) [inline service.fussballpopup/resources/language/English/strings.po]: height

File: service.fussballpopup/resources/language/English/strings.po [hunk 18]
```diff
@@ -0,0 +1,265 @@
...
+msgctxt "#30013"
+msgid "Advanced Settings"
+msgstr ""
+
+msgctxt "#30014"
+msgid "Advances Sttings(Only if you know what you are doing)"
+msgstr ""
+
+msgctxt "#30015"
+msgid "Fontt"
```
Comment (enen92) [inline service.fussballpopup/resources/language/English/strings.po]: font

File: service.fussballpopup/resources/language/English/strings.po [hunk 19]
```diff
@@ -0,0 +1,265 @@
...
+msgctxt "#30012"
+msgid "Old Messages"
+msgstr ""
+
+msgctxt "#30013"
+msgid "Advanced Settings"
+msgstr ""
+
+msgctxt "#30014"
+msgid "Advances Sttings(Only if you know what you are doing)"
```
Comment (enen92) [inline service.fussballpopup/resources/language/English/strings.po]: advanced settings

File: service.fussballpopup/resources/language/English/strings.po [hunk 20]
```diff
@@ -0,0 +1,265 @@
...
+msgid "Shows Football Games Messages Feed in Overlay WIndows"
+msgstr ""
+
+msgctxt "Addon Description"
+msgid "Shows Fussball Feed in Overlay WIndows"
+msgstr ""
+
+
+msgctxt "#30001"
+msgid "generel"
```
Comment (enen92) [inline service.fussballpopup/resources/language/English/strings.po]: general

File: service.fussballpopup/resources/language/English/strings.po [hunk 21]
```diff
@@ -0,0 +1,265 @@
...
+"Content-Transfer-Encoding: 8bit\n"
+"Language: en\n"
+"Plural-Forms: nplurals=2; plural=(n != 1)\n"
+
+msgctxt "Addon Summary"
+msgid "Shows Football Games Messages Feed in Overlay WIndows"
+msgstr ""
+
+msgctxt "Addon Description"
+msgid "Shows Fussball Feed in Overlay WIndows"
```
Comment (enen92) [inline service.fussballpopup/resources/language/English/strings.po]: I don't think those are used but, football :)

File: service.fussballpopup/service.py [hunk 22]
```diff
@@ -0,0 +1,475 @@
...
+   inhalt = urllib2.urlopen(req).read()   
+   return inhalt
+   
+def debug(content):
+    log(content, xbmc.LOGDEBUG)
+    
+def notice(content):
+    log(content, xbmc.LOGNOTICE)
+
+def log(msg, level=xbmc.LOGNOTICE):
```
Comment (enen92) [inline service.fussballpopup/service.py]: move the default to debug please

File: service.fussballpopup/service.py [hunk 23]
```diff
@@ -0,0 +1,475 @@
...
+
+if not xbmcvfs.exists(temp):
+   xbmcvfs.mkdirs(temp)
+
+
+
+def addDir(name, url, mode, iconimage, desc=""):
+    u = sys.argv[0]+"?url="+urllib.quote_plus(url)+"&mode="+str(mode)
+    ok = True
+    liz = xbmcgui.ListItem(name, iconImage=icon, thumbnailImage=iconimage)
```
Comment (enen92) [inline service.fussballpopup/service.py]: iconimage and thumbanailimage both deprecated -> setart

File: service.fussballpopup/service.py [hunk 24]
```diff
@@ -0,0 +1,475 @@
...
+    liz.setInfo(type="Video", infoLabels={"Title": name, "Plot": desc})  
+    if not iconimage or iconimage==icon or iconimage==defaultThumb:
+        iconimage = defaultBackground
+        liz.setProperty("fanart_image", iconimage)
```
Comment (enen92) [inline service.fussballpopup/service.py]: define this one in setArt as well

File: service.fussballpopup/service.py [hunk 25]
```diff
@@ -0,0 +1,475 @@
...
+    else:
+        liz.setProperty("fanart_image", defaultBackground)
+    ok = xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]), url=u, listitem=liz, isFolder=True)
+    return ok
+  
+def addLink(name, url, mode, iconimage, duration="", desc="", genre=''):
+    u = sys.argv[0]+"?url="+urllib.quote_plus(url)+"&mode="+str(mode)
+    ok = True
+    liz = xbmcgui.ListItem(name, iconImage=defaultThumb, thumbnailImage=iconimage)
```
Comment (enen92) [inline service.fussballpopup/service.py]: same

File: service.fussballpopup/service.py [hunk 26]
```diff
@@ -0,0 +1,475 @@
...
+    liz.setInfo(type="Video", infoLabels={"Title": name, "Plot": desc, "Genre": genre})
+    liz.setProperty('IsPlayable', 'true')
+    liz.addStreamInfo('video', { 'duration' : duration })
+    liz.setProperty("fanart_image", iconimage)
```
Comment (enen92) [inline service.fussballpopup/service.py]: same

File: service.fussballpopup/service.py [hunk 27]
```diff
@@ -0,0 +1,475 @@
...
+      greyoutlist=[]
+      lesezeitlist=[]
+      timelist=[] 
+      ids=[]
+      ins=[]
+      auss=[]
+      anzal_meldungen=[]        
+      foto1=""      
+      foto2=""      
+      xbmc.log("Hole Umgebung")
```
Comment (enen92) [inline service.fussballpopup/service.py]: please make debug strings in english

**Comment by enen92** [comment]
You also seem to have a lot of repeated code in service.py and menu.py. Any reason for that? Probably you can move the common stuff to a separated file which is used by the 2 scripts

File: service.fussballpopup/addon.xml [hunk 28]
```diff
@@ -0,0 +1,24 @@
...
+    <extension point="xbmc.service" library="service.py" start="login" />
+	  <extension point="xbmc.python.pluginsource" library="menu.py" >
+	    <provides>executable</provides>
+    </extension>
+    <extension point="xbmc.addon.metadata">
+    <summary lang="en">Shows Football Games Messages Feed in Overlay WIndows</summary>
+    <summary lang="de">Zeigt Fussball SPiele Nachrichten Updates in PupupFensterann</summary>
+    <description lang="en">Shows Fussball Feed in Overlay WIndows</description>
+    <description lang="de">Zeigt Fussball Updates in PupupFensterann</description>
+    <website></website>
```
Comment (enen92) [inline service.fussballpopup/addon.xml]: please add the licence tag

**Comment by enen92** [comment]
ping @tyl0re

**Comment by tyl0re** [comment]
Hopfully nothing sliped through. All should be corrected

**Pull request merged**